Adds initial guidelines for security reviewers (#5724)
department-of-veterans-affairs/va.gov-team: platform/security/hosting-a-security-review.md

# Hosting a security review

**Audience**: VSP security reviewers

**Context**: details and process for hosting a security review

## Philosophy

The goal of the security review is to ask, "what ways can this system break? How can it be misused?" By nature, a security review defies a concrete checklist.

The goal of this list of questions is to provide starting points for relevant discussions, and to ensure that "obvious" considerations aren't forgotten or missed.

## Questions

This list of questions is _not_ intended to be comprehensive. It _is_ intended to be comprehensively covered in a review.

### Privacy

#### Logs

> * What new log statements have the developers added? (FE or BE). This includes data sent to Sentry, Cloudwatch, Google Analytics, ...?

Reviewer: ```suggestion
* Are there any new logging statements? (FE or BE). This includes data sent to Sentry, Cloudwatch, Google Analytics, ...?
```

> * What kinds of information are being logging? Is there anything that uniquely identifies a user (PII)?

Reviewer: ```suggestion
* What kinds of information are logged in the new logging? Is there anything that uniquely identifies a user (Personally Identifiable Information)?
```

#### User tracking / identity

* Are there any new cookies?
* Are there any new uses of existing cookies?

### Security

* How are systems authenticating with one another?

> * Is anyone bypassing any obvious built-in protections? (CSRF protection in form submissions, auto-escaping in FE template rendering systems, use of `innerHTML`)

Reviewer: We need to train FE on these
